lambda-functions/database-factory-horizontal-scaling/main.go

```diff
@@ -0,0 +1,331 @@
+		return errors.Errorf("Environment variable MattermostNotificationsHook was not set")
+	}
+	if os.Getenv("MattermostAlertsHook") == "" {
+		return errors.Errorf("Environment variable MattermostAlertsHook was not set")
+	}
+	return nil
+}
+
+func getVPCs() ([]string, error) {
+	var next *string
+	sess, err := session.NewSession(&aws.Config{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get initiate AWS session")
+	}
+
+	client := ec2.New(sess)
+	filter := []*ec2.Filter{
+		{
+			Name:   aws.String("tag:Purpose"),
+			Values: []*string{aws.String("provisioning")},
+		},
+		{
+			Name:   aws.String("tag:Terraform"),
+			Values: []*string{aws.String("true")},
+		},
+		{
+			Name:   aws.String("tag:Available"),
+			Values: []*string{aws.String("false")},
+		},
+	}
+	var vpcList []string
+	for {
+		vpcs, err := client.DescribeVpcs(&ec2.DescribeVpcsInput{Filters: filter, NextToken: next})
+		if err != nil {
+			return nil, errors.Wrapf(err, "Unable to get Provisioning VPCs")
+		}
+
+		for _, vpc := range vpcs.Vpcs {
+			vpcList = append(vpcList, *vpc.VpcId)
+		}
+
+		if vpcs.NextToken == nil || *vpcs.NextToken == "" {
+			break
+		}
+
+		next = vpcs.NextToken
+
+	}
+	return vpcList, nil
+
+}
+
+func checkDBClustersScaling(vpcList []string) ([]string, error) {
+	sess, err := session.NewSession(&aws.Config{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get initiate AWS session")
+	}
+	client := resourcegroupstaggingapi.New(sess)
+
+	for _, vpc := range vpcList {
+		log.Infof("Checking DB clusters for VPC (%s)", vpc)
+		dbClusters, err := client.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+			TagFilters: []*resourcegroupstaggingapi.TagFilter{
+				{
+					Key:    aws.String("DatabaseType"),
+					Values: []*string{aws.String(os.Getenv("RDSMultitenantDBClusterTagDatabaseType"))},
+				},
+				{
+					Key:    aws.String("VpcID"),
+					Values: []*string{aws.String(vpc)},
+				},
+				{
+					Key:    aws.String("Purpose"),
+					Values: []*string{aws.String(os.Getenv("RDSMultitenantDBClusterTagPurpose"))},
+				},
+			},
+			ResourceTypeFilters: []*string{aws.String("rds:cluster")},
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get available multitenant DB clusters")
+		}
+		if len(dbClusters.ResourceTagMappingList) > 0 {
+			bool, err := checkVPCScaling(dbClusters, vpc)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to check if cluster scaling is needed")
+			}
+			if bool {
+				log.Infof("Initiating new RDS cluster deployment in vpc (%s)", vpc)
+				err = requestDeployCluster(vpc)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to request a new RDS cluster deployment")
+				}
+			}
+		}
+
+	}
+
+	return nil, nil
+}
+
+func requestDeployCluster(vpc string) error {
+	applyBool, err := strconv.ParseBool(os.Getenv("TerraformApply"))
+	if err != nil {
+		return errors.Wrap(err, "failed to parse bool from TerraformApply string")
+	}
```

Comment:
Can we move this check up to the `checkEnvVariables()` function? It would be better to fail early if we have bad values set.

Reply:
Yeap moved the check up to catch it early in the process. Great suggestion :)